src/screens/Details/TrackList/index.tsx

```diff
@@ -69,19 +100,78 @@ type TrackListProps = {
 };
 
 function TrackList({ trackInfo, HeaderComponent, onRefresh }: TrackListProps) {
+  const { spotifyAuth } =
+    useContext<SpotifyAuthContextData>(SpotifyAuthContext);
   const { sampleData, loading } = useWhoSampledAPI(trackInfo);
-  return (
-    <FlatList
-      refreshControl={
-        <RefreshControl onRefresh={onRefresh} refreshing={loading} />
-      }
-      ListHeaderComponent={HeaderComponent}
-      ListEmptyComponent={
-        loading ? <WhoSampledSkeleton /> : <EmptyListMessage />
+  const [selectedTrackIndex, setSelectedTrackIndex] = useState<number>(-1);
+
+  const [showSnackbar, setShowSnackbar] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+
+  useEffect(() => {
+    async function findAndQueueTrack(selectedTrack: WhoSampledData) {
+      const { track_name, artist } = selectedTrack;
+      const url = generateSpotifyTrackAndArtistQueryURL(track_name, artist);
+
+      try {
+        const { data } = await spotifyGETData(url, spotifyAuth);
+        const { items } = data.tracks;
+        const matchingTrack = findMatchingTrack(items, selectedTrack);
+        if (!matchingTrack) {
+          setSnackbarText(`Unable to find ${track_name} in search results`);
+          setShowSnackbar(true);
+          return;
+        }
+
+        const { uri } = matchingTrack;
+        const { status } = await spotifyPOSTData(
+          `v1/me/player/queue?uri=${uri}`,
+          spotifyAuth,
+        );
+        const queueResultText =
+          status === 204
+            ? `Queued ${track_name} by ${artist}`
+            : `Unable to queue track ${track_name}`;
+        setSnackbarText(queueResultText);
+      } catch (err) {
+        setSnackbarText(JSON.stringify(err));
       }
-      data={sampleData?.samples}
-      renderItem={TrackItem}
-    />
+      setShowSnackbar(true);
```

Comment:
Do a quick test to determine if on a reload we double queue, if so let's reset the index back to -1 right when we set the snackbar text

Reply:
Pulling on refresh does double queue, fixed this by setting the index selection to -1